components/scream/scripts/scripts_ctest_driver.py

```diff
@@ -0,0 +1,81 @@
+from utils import run_cmd, expect, check_minimum_python_version
+from machines_specs import is_machine_supported, setup_mach_env
+
+check_minimum_python_version(3, 4)
+
+import os, shutil
+from pathlib import Path
+
+###############################################################################
+class ScriptsCtestDriver(object):
+###############################################################################
+
+    ###########################################################################
+    def __init__(self, submit=False, machine=None, root_dir=None, work_dir=None, dry_run=False):
+    ###########################################################################
+
+        self._submit   = submit
+        self._machine  = machine
+        self._root_dir = root_dir
+        self._work_dir = work_dir
+        self._dry_run  = dry_run
+
+        # Probe machine if none was specified
+        if self._machine is None:
+            # We could potentially integrate more with CIME here to do actual
+            # nodename probing.
+            if "CIME_MACHINE" in os.environ and is_machine_supported(os.environ["CIME_MACHINE"]):
+                self._machine = os.environ["CIME_MACHINE"]
+            else:
+                expect(False,
+                       "scripts-ctest-driver requires either the machine arg or CIME_MACHINE in env")
+
+        # Compute root dir (where repo is) and work dir (where build/test will happen)
+        if not self._root_dir:
+            self._root_dir = Path(__file__).resolve().parent.parent
+        else:
+            self._root_dir = Path(self._root_dir).resolve()
+            expect(self._root_dir.is_dir() and self._root_dir.parts()[-2:] == ('scream', 'components'),
+                   "Bad root-dir '{}', should be: $scream_repo/components/scream".format(self._root_dir))
+
+        if self._work_dir is None:
+            self._work_dir = self._root_dir.absolute().joinpath("ctest-build-scripts")
+        else:
+            self._work_dir = Path(self._work_dir).absolute()
+
+        if self._work_dir.exists():
+            expect(self._work_dir.is_dir(), "Work dir {} exists but is not a directory".format(self._work_dir))
+            shutil.rmtree(str(self._work_dir))
+
+        self._work_dir.mkdir(parents=True)
+
+        setup_mach_env(self._machine)
+
+    ###############################################################################
+    def generate_ctest_config(self, extra_configs):
+    ###############################################################################
+        result = "CTEST_PARALLEL_LEVEL=1 CIME_MACHINE={} ".format(self._machine)
```

Comment:
Here too: no point in setting CTEST_PARALLEL_LEVEL=1, since we later hard code `-j1` anyways.